airflow-core/src/airflow/models/dag.py

```diff
@@ -463,6 +464,15 @@ def validate_executor_field(self):
                         "update the executor configuration for this task."
                     )
 
+    def get_dagrun_creation_deadlines(self) -> DeadlineAlert | None:
+        """If the DAG has a deadline related to DagRun, return it; else return None."""
+        if (
+            not (deadline := self.deadline)
+            or type(deadline.reference) not in DeadlineReference.TYPES.DAGRUN_CREATED
```

Comment:
Is this better?

```suggestion
            or not isinstance(deadline.reference, tuple(DeadlineReference.TYPES.DAGRUN_CREATED))
```

Reply:
Oh, I was actually looking at that last night.  I'm going to switch that `if` around, it's a bit cleaner as:

```
      if (deadline := self.deadline) and type(deadline.reference) in DeadlineReference.TYPES.DAGRUN_CREATED:
            return deadline
        return None
```

For `type(foo) in bar` vs `isinstance(foo, tuple(bar))`, I kinda like the easier readability of the first but don't feel strongly.  I'll cast the TYPES to a tuple in their definition and switch to a more simplified version of your suggestion.

Reply:
Nevermind, I'll just make the change.